drivers/stepper/Kconfig

```diff
@@ -0,0 +1,25 @@
+# Stepper controller driver configuration options
+
+# Copyright (c) 2020 Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig STEPPER
```

Comment:
Could you add [EXPERIMENTAL] ?

It's a really good API addition, but once it will be in master (hopefully for 2.4), users might have some improvements, changes etc... Could be a feature for position reference (homing) or else. 

Having it specifically set "experimental" will avoid to go through a tedious process of changing a stable API

Reply:
good point, marked as experimental. thanks!

Reply:
The API lifecycle doesn't require this, though perhaps it's something that could be added @carlescufi 

APIs start out as experimental and don't become stable until [certain conditions](https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#stable) are met including being available for two development releases.  A first step that seems to have been missed is adding this to the API project so it can be triaged and tracked.  I've done that.

Reply:
Indeed, I am following the legacy rule. Which imo, could be kept as well. It's more obvious to users that the API is not stable if it's marked experimental in Kconfig.

Reply:
Yes, I will make a note to add this to the guide, thanks @pabigot.

Reply:
Also to discuss this (Kconfig marking of EXPERIMENTAL) in API meeting.  It looks like a lot of Zephyr subsystems are doing this for features within  a subsystem, but possibly not for the entire API.  (Except for a couple like LoRa and Modem which aren't even listed in the API overview.)